### ISSUE TITLE
Create a function that allows a single selection from a list of strings

### ISSUE THREAD

**Issue by nazmulidris** [r3bl-org/r3bl_tuify#1]
Create this function to allow an interactive terminal session w/ a user to display and select a single item that they choose.

```rust
pub fn select_from_list(
  list: Vec<String>,
  max_rows: usize,
  selection_mode: SelectionMode,
) -> Result<Vec<String>>

pub enum SelectionMode {
  SingleSelect, MultiSelect
}
```

The `max_rows` argument specifies how many rows will be taken up in the terminal window. 

1. If there are more items in the `list` than this number, then the rendered list will scroll up and down. In this case, need to display the number of items above the top of the rendered list, and below the bottom of the rendered list.
3. If there are fewer items in the list than this number, then the `max_rows` will be the size of the `list`.

Raw mode will be used in order to accept user input. And we listen for the following key events:
1. Up -> move the caret up, might scroll
2. Down -> move the caret down, might scroll
4. Enter -> choose the selected item and exit raw mode
5. Esc -> cancel the selection and exit raw mode

Finally a `Result<Vec<String>>` is returned by this function which can hold one or more chosen items. For now only `SelectionMode::SingleSelect` is supported.

**Comment by nazmulidris** [comment]
https://github.com/r3bl-org/r3bl_tuify/assets/2966499/7da6f94a-0120-4562-9e44-53cd49a26aee

**Comment by nazmulidris** [comment]
Closed by bc0a2f0
